### ISSUE TITLE
[Suggestion] Remove "Insert Item.." from context menu in the empty editor

### ISSUE THREAD

**Issue by v-mengwe** [microsoft/vscode-azurearmtools#885]
**OS:** Win10
**Build Version:** 20200729.4

**Repro Steps:**

1. In VS Code, create an empty "Azure Resource Manager Template" file.
2. Right click in the editor --> Insert Item... --> Select "Function".
3. Check the result.

**Actual:**
There is an error "Invalid ARM template".
![image](https://user-images.githubusercontent.com/62053665/88785235-86494c80-d1c3-11ea-9229-ae9ea2d5e445.png)

**Suggestion:**
Remove "Insert Item.." from context menu in the empty editor.

**More info:**

1. This issue also reproduces when inserting "Parameter/Output/Resource/Variable".
2. This issue  does not reproduce when adding following code to this "Azure Resource Manager Template" file:
 ```
{ 
 "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#"
 }
 ```

**Comment by StephenWeatherford** [comment]
The error should come earlier if at all, and it should be more friendly.

**Comment by jodavis** [comment]
We can add the basic template content to an empty file, prior to adding the first resource using this command.

**Comment by StephenWeatherford** [comment]
Improved the error message
Fixed by #1025
